python/doc/examples/data_analysis/distribution_fitting/plot_advanced_mle_estimator.py

```diff
@@ -0,0 +1,289 @@
+# :math:`\gamma` is zero.
+# Hence, the estimator may not necessarily lead to an estimated parameter
+# exactly equal to zero.
+# In this example, we assume that we know that this parameter is zero.
+# In this case, the simplest method is to use the
+# :meth:`~openturns.MaximumLikelihoodFactory.setKnownParameter` method.
+# Furthermore, we know that the absolute value of the mean of the underlying normal
+# distribution cannot be greater than 5.
+# Finally, we know that the standard deviation of the underlying
+# normal distribution cannot be lower than, say, :math:`\epsilon = 10^{-4}`.
+
+# %%
+logNormalFactoryWithZeroLocation = ot.MaximumLikelihoodFactory(standardLogNormal)
+logNormalFactoryWithZeroLocation.setKnownParameter([0.0], [2])
+lowerBound = [-5.0, 1.0e-4]
+upperBound = [5.0, -1.0]
+finiteLowerBound = [True, True]
+finiteUpperBound = [True, False]
+bounds = ot.Interval(lowerBound, upperBound, finiteLowerBound, finiteUpperBound)
+logNormalFactoryWithZeroLocation.setOptimizationBounds(bounds)
+fittedDistribution = logNormalFactoryWithZeroLocation.build(sample)
+print("   fittedDistribution = ", fittedDistribution)
+
+# %%
+# See the list of solvers compatible with a given MLE problem
+# -----------------------------------------------------------
+
+# %%
+# In the next script, we perform a loop over the algorithms
+# from the :class:`NLopt` class and select the algorithms
+# which can solve the problem.
+# This is done using `try/except` statements.
+# Moreover, some solvers pretend to solve the problem, but do
+# not make any change to the parameters.
+# In this case, we do not select them.
+
+# %%
+defaultParameter = standardLogNormal.getParameter()
+defaultSolver = logNormalFactoryWithZeroLocation.getOptimizationAlgorithm()
+startingPoint = defaultSolver.getStartingPoint()
+print("Default solver = ", defaultSolver.getImplementation().getClassName())
+availableSolverList = []
+differenceThreshold = 1.0e-8
+for algorithmName in ot.NLopt.GetAlgorithmNames():
+    print("- ", algorithmName)
+    solver = ot.NLopt(algorithmName)
+    solver.setStartingPoint(startingPoint)
+    logNormalFactoryWithZeroLocation.setOptimizationAlgorithm(solver)
+    try:
+        fittedDistribution = logNormalFactoryWithZeroLocation.build(sample)
+        print("   fittedDistribution = ", fittedDistribution)
+        parameter = fittedDistribution.getParameter()
+        differenceOfParameters = parameter - defaultParameter
+        if differenceOfParameters.norm() > differenceThreshold:
+            # Consider only solvers which make a difference after optimization
+            availableSolverList.append(algorithmName)
+    except TypeError:
+        print("   Fail : Impossible to build with ", algorithmName)
+
+# %%
+print("Available solvers = ", len(availableSolverList))
+for name in availableSolverList:
+    print("- ", name)
+
+# %%
+# Pick one and try it.
+
+# %%
+algorithmName = "LD_SLSQP"
```

Comment:
the example must be disabled when nlopt is, see python/test/CMakeLists.txt at line 841

Reply:
Thank you for the tip. Fixed at ffee0930ead4ce449d045d877577cd1ff7335e78